src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -410,267 +584,458 @@ private string GenerateForObject(TypeMetadata typeMetadata)
 
-                // Add imports to root type.
-                usingStatements.Add(FormatAsUsingStatement(typeMetadata.Type.Namespace));
+                sb.Append(@$"{GetJsonContextDeclarationSource()}
+{{");
 
-                switch (typeMetadata.ClassType)
+                foreach (string propName in _currentContext.RuntimePropertyNames)
                 {
-                    case ClassType.Nullable:
-                        {
-                            AddUsingStatementsForType(typeMetadata.NullableUnderlyingTypeMetadata!);
-                        }
-                        break;
-                    case ClassType.Enumerable:
-                        {
-                            AddUsingStatementsForType(typeMetadata.CollectionValueTypeMetadata);
-                        }
-                        break;
-                    case ClassType.Dictionary:
-                        {
-                            AddUsingStatementsForType(typeMetadata.CollectionKeyTypeMetadata);
-                            AddUsingStatementsForType(typeMetadata.CollectionValueTypeMetadata);
-                        }
-                        break;
-                    case ClassType.Object:
-                        {
-                            if (typeMetadata.PropertiesMetadata != null)
-                            {
-                                foreach (PropertyMetadata metadata in typeMetadata.PropertiesMetadata)
-                                {
-                                    AddUsingStatementsForType(metadata.TypeMetadata);
-                                }
-                            }
-                        }
-                        break;
-                    default:
-                        break;
+                    sb.Append($@"
+    private static {JsonEncodedTextTypeRef} {propName}PropName = {JsonEncodedTextTypeRef}.Encode(""{propName}"");");
                 }
 
-                void AddUsingStatementsForType(TypeMetadata typeMetadata)
-                {
-                    usingStatements.Add(FormatAsUsingStatement(typeMetadata.Type.Namespace));
+                sb.Append(@"
+}");
 
-                    if (typeMetadata.CollectionKeyTypeMetadata != null)
-                    {
-                        Debug.Assert(typeMetadata.CollectionValueTypeMetadata != null);
-                        usingStatements.Add(FormatAsUsingStatement(typeMetadata.CollectionKeyTypeMetadata.Type.Namespace));
-                    }
+                return sb.ToString();
+            }
 
-                    if (typeMetadata.CollectionValueTypeMetadata != null)
-                    {
-                        usingStatements.Add(FormatAsUsingStatement(typeMetadata.CollectionValueTypeMetadata.Type.Namespace));
-                    }
-                }
+            private string GetJsonContextDeclarationSource() => $"internal partial class {_currentContext.ContextType.Name} : {JsonSerializerContextTypeRef}";
```

Comment:
If the developer says `public partial class JsonContext : JsonSerializerContext`, this will result in a build error:

`Partial declarations of 'JsonContext' have conflicting accessibility modifiers`.

We should let developers decide the visibility of the class.

Reply:
Addressed in https://github.com/dotnet/runtime/pull/53212/commits/2dbebc9df14dbc7c966be22f6a75db5384af8829.